image/image.js

```diff
@@ -1,14 +1,37 @@
 module.exports = function(RED) {
   function ImageNode(config) {
     RED.nodes.createNode(this, config);
+    this.active = (config.active === null || typeof config.active === "undefined") || config.active;
     var node = this;
 
     node.on("input", function(msg) {
-      RED.comms.publish("image", {
-        id: this.id,
-        data: msg.payload.toString("base64")
-      });
+      if (this.active) {
+        RED.comms.publish("image", {
+          id: this.id,
+          data: msg.payload.toString("base64")
+        });
+      }
     });
   }
+  
+  // Via the button on the node (in the flow editor), the image pushing can be enabled or disabled
+  RED.httpAdmin.post("/image-output/:id/:state", RED.auth.needsPermission("image-output.write"), function(req,res) {
+    var node = RED.nodes.getNode(req.params.id);
+    var state = req.params.state;
+    if (node !== null && typeof node !== "undefined" ) {
+        if (state === "enable") {
+            node.active = true;
+            res.sendStatus(200);
```

Comment:
I think in both cases of state, the response status should be 200. 201 is commonly used to indicate a new resource was created, which isn't the case here.  Could we use some other way of distinguishing the proceeding state on the frontend code?

Reply:
Hi Riku,

A little bit later as expected, but it's a busy week ...
As I mentioned in the beginning, I based this pull request on the Debug node.  Hadn't been reviewing the Debug code in depth, since it is written by the core team.

But indeed you are right.  In the latest version, I just use send('deactived') to let the flow editor know that the server side node has been deactived.  In that case the status is automatically set to 200:

![image](https://user-images.githubusercontent.com/14224149/36815749-ffe54048-1cdb-11e8-8973-a0c77c7767ac.png)

This makes the html also simple: I just show the "deactivated" (or "activated") immediately in the popup notification.